autoreadme.go

```diff
@@ -207,6 +217,52 @@ func getDoc(dir string) (Doc, error) {
 	}, nil
 }
 
+func renderExamples(bi *build.Package) (map[string]string, error) {
+	filter := func(fi os.FileInfo) bool {
+		if fi.IsDir() {
+			return false
+		}
+		nm := fi.Name()
+		for _, f := range append(bi.TestGoFiles, bi.XTestGoFiles...) {
+			if nm == f {
+				return true
+			}
+		}
+		return false
+	}
+	pkgs, err := parser.ParseDir(token.NewFileSet(), bi.Dir, filter, parser.ParseComments)
+	if err != nil {
+		return nil, err
+	}
+	pkg := pkgs[bi.Name]
```

Comment:
I guess this is broken if this is a test from `XTestGoFiles`. Will check.

Reply:
Yup, broken. I can merge the tests from `yourpackage` and `yourpackage_test`, but there might be collisions, and it's non-deterministic which one you will get. Another option would be to _only_ add tests from `XTestGoFiles`, which are generally the ones which are nicer for copy-paste anyway.